### ISSUE TITLE
Cannot get signal in MDF4 because of error in dependency_list

### ISSUE THREAD

**Issue by JulienGrv** [danielhrisca/asammdf#21]
Hi everyone,

When I try to get a list of signals that belongs to the same ChannelArrayBlock, the last signal has its channel_dependencies not empty (although it is a normal Channel).
The dependency_list being not empty, it tries to get every signals like if it was a ChannelArrayBlock. This signal being itself in the dependency_list, we enter into an infinite loop and asammdf crashes when reaching Python maximal recursion limit.

When looking a bit more in detail, the dependency_list that we get for this signal, is actually the dependency_list of the ChannelArrayBlock it belongs to.

In the ChannelArrayBlock I tested, I have 3 signals, so when I put an offset of three, the get function is working again:
```
dependency_list = grp['channel_dependencies'][ch_nr + 3]
instead of:
dependency_list = grp['channel_dependencies'][ch_nr]
in get function
```

However, I am not sure yet if this offset is constant or if it depends.

I am still doing some investigation to find out why the channel_dependencies does not match with the channels we have.

**Comment by danielhrisca** [comment]
Hello Julien,

I'm not sure what this kind of recursion might mean; could you post a MDFValidator screenshot of this Channel and its ChannelDepenedencyBlock?

If you're allowed to send the file via e-mail it would be perfect for investigation.

**Comment by danielhrisca** [comment]
I will probably disappoint you: I often use _print_, _pprint_ and _hex_; most times this is enough if you know where to place them. We also have to thank _Vector Gmbh_ for _MDFValidator_. 
I should definitely learn to use the debuggers more.

**Comment by JulienGrv** [comment]
Ah ah, I am not disappointed at all. Good programmers always know directly where the error comes from, no need of a debugger ;) 

Anyway, I will use the MDFValidator to see if it gives me some more information to find my way in this big MDF file!

**Comment by JulienGrv** [comment]
In theory, I should have as many channels as the sum of the channels_dependencies for every groups?

In my case, there is 291081 channels in the channels_db. While the sum of channels_dependencies for every groups is 335095:

```python
lengths = [len(grp['channel_dependencies']) for grp in self.groups]
total = sum(lengths)
print(total)  # prints 335095
print(len(self.channels_db))  # prints 291081
```

**Comment by danielhrisca** [comment]
If you have repeating channel names then you could see a difference.

Try this:

``` python
print(len(self.channels_db))

lengths = [len(grp['channel_dependencies']) for grp in self.groups]
total = sum(lengths)
print(total)  # prints 335095

lengths = [len(grp['channels']) for grp in self.groups]
total = sum(lengths)
print(total) # should print 335095 as well

lengths = [len(occurrences) for occurrences in self.channels_db.values()]
total = sum(lengths)
print(total) # should print 335095 as well

for channel, occurrences in self.channels_db.items():
    if len(occurrences) > 1:
        print(channel, occurrences)
```

For the mf4 test file it prints:
```python
725
36424
36424
36424
t [(0, 0), (1, 0), (2, 0), (3, 0), (4, 0), (5, 0), (6, 0), (7, 0), (8, 0), (9, 0), (10, 0), (11, 0), (12, 0), (13, 0), (14, 0), (15, 0), (16, 0), (17, 0), (18, 0), (19, 0), (20, 0), (21, 0), (22, 0), (23, 0), (24, 0), (25, 0), (26, 0), (27, 0), (28, 0), (29, 0), (30, 0), (31, 0), (32, 0), (33, 0), (34, 0), (35, 0), (36, 0), (37, 0), (38, 0), (39, 0), (40, 0), (41, 0), (42, 0), (43, 0), (44, 0), (45, 0), (46, 0), (47, 0), (48, 0), (49, 0), (50, 0), (51, 0), (52, 0), (53, 0), (54, 0), (55, 0), (56, 0), (57, 0), (58, 0), (59, 0), (60, 0), (61, 0), (62, 0), (63, 0), (64, 0), (65, 0), (66, 0), (67, 0), (68, 0), (69, 0), (70, 0), (71, 0), (72, 0), (73, 0), (74, 0), (75, 0), (76, 0), (77, 0), (78, 0), (79, 0), (80, 0), (81, 0), (82, 0), (83, 0), (84, 0), (85, 0), (86, 0), (87, 0), (88, 0), (89, 0), (90, 0), (91, 0), (92, 0), (93, 0), (94, 0), (95, 0), (96, 0), (97, 0), (98, 0), (99, 0), (100, 0), (101, 0), (102, 0), (103, 0), (104, 0), (105, 0), (106, 0), (107, 0), (108, 0), (109, 0), (110, 0), (111, 0), (112, 0), (113, 0), (114, 0), (115, 0), (116, 0), (117, 0), (118, 0), (119, 0), (120, 0), (121, 0), (122, 0), (123, 0), (124, 0), (125, 0), (126, 0), (127, 0), (128, 0), (129, 0), (130, 0), (131, 0), (132, 0), (133, 0), (134, 0), (135, 0), (136, 0), (137, 0), (138, 0), (139, 0), (140, 0), (141, 0), (142, 0), (143, 0), (144, 0), (145, 0), (146, 0), (147, 0), (148, 0), (149, 0), (150, 0), (151, 0), (152, 0), (153, 0), (154, 0), (155, 0), (156, 0), (157, 0), (158, 0), (159, 0), (160, 0), (161, 0), (162, 0), (163, 0), (164, 0), (165, 0), (166, 0), (167, 0), (168, 0), (169, 0), (170, 0), (171, 0), (172, 0), (173, 0), (174, 0), (175, 0), (176, 0), (177, 0), (178, 0), (179, 0), (180, 0), (181, 0), (182, 0)]
C1 [(0, 1), (1, 1), (2, 1), (3, 1), (4, 1), (5, 1), (6, 1), (7, 1), (8, 1), (9, 1), (10, 1), (11, 1), (12, 1), (13, 1), (14, 1), (15, 1), (16, 1), (17, 1), (18, 1), (19, 1), (20, 1), (21, 1), (22, 1), (23, 1), (24, 1), (25, 1), (26, 1), (27, 1), (28, 1), (29, 1), (30, 1), (31, 1), (32, 1), (33, 1), (34, 1), (35, 1), (36, 1), (37, 1), (38, 1), (39, 1), (40, 1), (41, 1), (42, 1), (43, 1), (44, 1), (45, 1), (46, 1), (47, 1), (48, 1), (49, 1), (50, 1), (51, 1), (52, 1), (53, 1), (54, 1), (55, 1), (56, 1), (57, 1), (58, 1), (59, 1), (60, 1), (61, 1), (62, 1), (63, 1), (64, 1), (65, 1), (66, 1), (67, 1), (68, 1), (69, 1), (70, 1), (71, 1), (72, 1), (73, 1), (74, 1), (75, 1), (76, 1), (77, 1), (78, 1), (79, 1), (80, 1), (81, 1), (82, 1), (83, 1), (84, 1), (85, 1), (86, 1), (87, 1), (88, 1), (89, 1), (90, 1), (91, 1), (92, 1), (93, 1), (94, 1), (95, 1), (96, 1), (97, 1), (98, 1), (99, 1), (100, 1), (101, 1), (102, 1), (103, 1), (104, 1), (105, 1), (106, 1), (107, 1), (108, 1), (109, 1), (110, 1), (111, 1), (112, 1), (113, 1), (114, 1), (115, 1), (116, 1), (117, 1), (118, 1), (119, 1), (120, 1), (121, 1), (122, 1), (123, 1), (124, 1), (125, 1), (126, 1), (127, 1), (128, 1), (129, 1), (130, 1), (131, 1), (132, 1), (133, 1), (134, 1), (135, 1), (136, 1), (137, 1), (138, 1), (139, 1), (140, 1), (141, 1), (142, 1), (143, 1), (144, 1), (145, 1), (146, 1), (147, 1), (148, 1), (149, 1), (150, 1), (151, 1), (152, 1), (153, 1), (154, 1), (155, 1), (156, 1), (157, 1), (158, 1), (159, 1), (160, 1), (161, 1), (162, 1), (163, 1), (164, 1), (165, 1), (166, 1), (167, 1), (168, 1), (169, 1), (170, 1), (171, 1), (172, 1), (173, 1), (174, 1), (175, 1), (176, 1), (177, 1), (178, 1), (179, 1), (180, 1), (181, 1), (182, 1)]
...
C704 [(112, 704), (130, 704)]
```

**Comment by JulienGrv** [comment]
Fixed in PR #23
